woocommerce/woocommerce
# Introduce live branches UI page

### All Submissions:






### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/34524

This PR is mostly boilerplate required to get live branches UI work ready for others to contribute to. We don't have a feature flags setup for beta tester yet, so I've commented out the line that registers the menu item. You can uncomment it to see the menu item, but you can also visit the page manually to see the UI so far by visiting this url directly: `http://YOUR-LOCAL-WP-HOST/wp-admin/admin.php?page=wc-admin&path=%2Flive-branches`

Note: there are a few TS ignores of some type issues. We could address them all here, but since this is in a prototype stage my opinion is it doesn't matter till we build the real UI.

Introduced changes:

*  A webpack build with tooling for Prettier and TypeScript all set up.
* An API endpoint that loads the live branches manifest (required to get around CORs issues loading the manifest on front-end)
* Some basic React/TS code that loads manifest from the API endpoint and lists PRs from the manifest. This can be expanded on to support actual installing of live branches.



### How to test the changes in this Pull Request:


1. To build the front-end navigate to `plugins/woocommerce-beta-tester` and run `pnpm start`
2. Either uncomment this line: https://github.com/woocommerce/woocommerce/blob/4c12b494eccf1f6f2d8945e34c8a2a7164c3e200/plugins/woocommerce-beta-tester/includes/class-wc-beta-tester-live-branches.php#L24 or visit `http://YOUR-LOCAL-WP-HOST/wp-admin/admin.php?page=wc-admin&path=%2Flive-branches`
3. There should be a rendered list of prs extracted from the live branches manifest



### Other information:




### FOR PR REVIEWER ONLY:
